src/js/extensions/anchor.js

```diff
@@ -60,10 +60,9 @@ var AnchorForm;
             event.preventDefault();
             event.stopPropagation();
 
-            var selectedParentElement = Selection.getSelectedParentElement(Selection.getSelectionRange(this.document)),
-                firstTextNode = Util.getFirstTextNode(selectedParentElement);
+            var selectedParentElement = Selection.getSelectedParentElement(Selection.getSelectionRange(this.document));
```

Comment:
this should also handle the case where the start container of the selection is the anchor tag.